pkg/runtime/field_export_reconciler.go

```diff
@@ -0,0 +1,777 @@
+		return errors.New("cannot bind to field export. reconciler marked for reconciling AWS resources")
+	}
+
+	return ctrlrt.NewControllerManagedBy(
+		mgr,
+	).For(
+		// Read only field export objects
+		&ackv1alpha1.FieldExport{},
+	).WithEventFilter(
+		predicate.GenerationChangedPredicate{},
+	).Complete(r)
+}
+
+// BindControllerManagerForAWSResource sets up the AWSResourceReconciler with an
+// instance of an upstream controller-runtime.Manager, watching for changes in
+// AWS Resource CRs
+func (r *fieldExportReconciler) BindControllerManagerForAWSResource(mgr ctrlrt.Manager) error {
+	r.kc = mgr.GetClient()
+	r.apiReader = mgr.GetAPIReader()
+
+	if ackcompare.IsNil(r.rd) {
+		return errors.New("cannot bind to AWS resource. reconciler marked for reconciling field exports")
+	}
+
+	return ctrlrt.NewControllerManagedBy(
+		mgr,
+	).For(
+		r.rd.EmptyRuntimeObject(),
+	).WithEventFilter(
+		// Update on both status and spec changes
+		predicate.ResourceVersionChangedPredicate{},
+	).Complete(r)
+}
+
+// Reconcile implements `controller-runtime.Reconciler` and handles reconciling
+// a CR CRUD request
+func (r *fieldExportReconciler) Reconcile(ctx context.Context, req ctrlrt.Request) (ctrlrt.Result, error) {
+	return r.handleReconcileError(r.reconcile(ctx, req))
+}
+
+func (r *fieldExportReconciler) reconcile(ctx context.Context, req ctrlrt.Request) error {
+	// Determine if we are reconciling an ACK resource
+	if !ackcompare.IsNil(r.rd) {
+		return r.reconcileSourceResource(ctx, req)
+	}
+
+	// We are reconciling a field export CR
+	return r.reconcileFieldExport(ctx, req)
+}
+
+// reconcileFieldExport handles updates to `FieldExport` resources
+func (r *fieldExportReconciler) reconcileFieldExport(ctx context.Context, req ctrlrt.Request) error {
+	feObject, err := r.getFieldExport(ctx, req)
+	if err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	sourceGK := feObject.Spec.From.Resource.GroupKind
+	sourceName := types.NamespacedName{
+		Name: *feObject.Spec.From.Resource.Name,
+		// We only support pulling from resources in
+		// the same namespace
+		Namespace: req.Namespace,
+	}
+
+	if feObject.DeletionTimestamp != nil {
+		return r.cleanup(ctx, feObject)
+	}
+
+	// Check if the target API group matches with the controller
+	var controllerRMF acktypes.AWSResourceManagerFactory
+	for _, v := range r.sc.GetResourceManagerFactories() {
+		controllerRMF = v
+		break
+	}
+	if sourceGK.Group != controllerRMF.ResourceDescriptor().GroupKind().Group {
+		ackrtlog.DebugFieldExport(r.log, feObject, "target resource API group is not of this service. no-op")
+		return nil
+	}
+
+	if err := r.markManaged(ctx, feObject); err != nil {
+		return err
+	}
+
+	// Look up the rmf for the given target resource GVK
+	rmf, ok := (r.sc.GetResourceManagerFactories())[sourceGK.String()]
+	if !ok {
+		return ackerr.ResourceManagerFactoryNotFound
+	}
+
+	sourceObject, err := r.getSourceResource(ctx, rmf.ResourceDescriptor(), sourceName)
```

Comment:
It seems to me that because the controller's Role or ClusterRole is the thing that is reading the source resource, this would be an easy way for a Kubernetes user to subvert Kubernetes RBAC. If a Kubernetes user only has read and write permissions on the `FieldExport` Kind and does not have read permissions on the source resource Kind, the functionality here essentially allows them to read any information (by specifying a path on that CRD) on that source resource Kind.